underfs/s3a/src/test/java/alluxio/underfs/s3a/S3AUnderFileSystemTest.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.underfs.s3a;
+
+import alluxio.AlluxioURI;
+
+import com.amazonaws.AmazonClientException;
+import com.amazonaws.services.s3.AmazonS3Client;
+import com.amazonaws.services.s3.model.ListObjectsV2Request;
+import com.amazonaws.services.s3.transfer.TransferManager;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Matchers;
+import org.mockito.Mockito;
+
+import java.io.IOException;
+
+/**
+ * Unit tests for the {@link S3AUnderFileSystem}.
+ */
+public class S3AUnderFileSystemTest {
+  private S3AUnderFileSystem mS3UnderFileSystem;
+  private AmazonS3Client mClient;
+  private TransferManager mManager;
+
+  private static final String PATH = "path";
+  private static final String SRC = "src";
+  private static final String DST = "dst";
+
+  private static final String BUCKET_NAME = "bucket";
+  private static final String BUCKET_PREFIX = "prefix";
+  private static final short BUCKET_MODE = 0;
+  private static final String ACCOUNT_OWNER = "account owner";
+  private static final String ACCOUNT_OWNER_ID = "account owner id";
+
+  /**
+   * Set up.
+   */
+  @Before
+  public void before() throws InterruptedException, AmazonClientException {
+    mClient = Mockito.mock(AmazonS3Client.class);
+    mManager = Mockito.mock(TransferManager.class);
+    mS3UnderFileSystem = new S3AUnderFileSystem(new AlluxioURI(""), mClient, BUCKET_NAME,
+            BUCKET_PREFIX, BUCKET_MODE, ACCOUNT_OWNER, ACCOUNT_OWNER_ID, mManager);
+  }
+
+  /**
+   * Test case for {@link S3AUnderFileSystem#delete(String, boolean)}.
+   * when {@link AmazonClientException} is thrown.
```

Comment:
If we add "AmazonClientException" to the test name we probably don't need this javadoc, same throughout this PR